src/rateEngine/RateElement.ts

```diff
@@ -9,29 +9,85 @@ import LoadProfile from './LoadProfile';
 import ValidatorFactory from './ValidatorFactory';
 import { RateComponentInterface } from './RateComponent';
 import { Error } from './validators/_Validator';
+import SurchargeAsPercent from './billingDeterminants/SurchargeAsPercent';
 
 export interface RateElementInterface {
+  id?: string;
   rateElementType: RateElementType;
-  rateComponents: Array<RateComponentInterface & BillingDeterminantFactoryInterface>;
+  rateComponents?: Array<RateComponentInterface & BillingDeterminantFactoryInterface>;
   name: string;
+  billingCategory?: BillingCategory;
+  rateElementIds?: Array<string>;
+  percent?: number;
+  appliesToAll ?: boolean;
```

Comment:
This is probably my least favorite part of the changes... `rateElementIds` and `appliesToAll` could apply to any future type of rate element that needs to be aware of other rate elements, but `percent` is pretty specific to the `SurchargeAsPercent` rate element. I could see this interface getting bloated if future rate elements also have unique parameters.

The counterargument is that this exact situation exists for the `RateComponentInterface`, and TypeScript allows to handle that elegantly. So maybe this is fine, even for the future.